wolvic/browser/webdata_services/web_data_service_factory.cc

```diff
@@ -0,0 +1,75 @@
+#include "wolvic/browser/webdata_services/web_data_service_factory.h"
+
+#include "base/files/file_path.h"
+#include "base/functional/bind.h"
+#include "base/no_destructor.h"
+#include "components/webdata_services/web_data_service_wrapper.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/browser_task_traits.h"
+#include "content/public/browser/browser_thread.h"
+
+// This class is derived from //chrome/webdata_service/WebDataServiceFactory.
+
+namespace wolvic {
+
+namespace {
+
+// Callback to show error dialog on context load error.
+void ContextErrorCallback(WebDataServiceWrapper::ErrorType error_type,
+                          sql::InitStatus status,
+                          const std::string& diagnostics) {
+  // TODO(jfernandez): Implement this error management callback.
+}
+
+std::unique_ptr<KeyedService> BuildWebDataService(
+    content::BrowserContext* context) {
+  const base::FilePath& path = context->GetPath();
+  return std::make_unique<WebDataServiceWrapper>(
+      path, "" /* application locale */ ,
+      content::GetUIThreadTaskRunner({}),
+      base::BindRepeating(&ContextErrorCallback));
+}
+
+} // namespace
+
+WebDataServiceFactory::WebDataServiceFactory() = default;
+
+WebDataServiceFactory::~WebDataServiceFactory() = default;
+
+// static
+WebDataServiceWrapper* WebDataServiceFactory::GetForContext(
+    content::BrowserContext* context,
+    ServiceAccessType access_type) {
+  return GetForBrowserContext(context, access_type);
+}
+
+// static
+WebDataServiceWrapper* WebDataServiceFactory::GetForContextIfExists(
+    content::BrowserContext* context,
+    ServiceAccessType access_type) {
+  return GetForBrowserContextIfExists(context, access_type);
+}
+
+// static
+WebDataServiceFactory* WebDataServiceFactory::GetInstance() {
+  static base::NoDestructor<WebDataServiceFactory> instance;
+  return instance.get();
+}
+
+content::BrowserContext* WebDataServiceFactory::GetBrowserContextToUse(
+    content::BrowserContext* context) const {
+  // Create a separate instance of the service for the Incognito context.
```

Comment:
Can't we do this in this patch?

Reply:
It depends on some Profile logic implemented in chorme, which redirects the BrowserContext to a new instance created for the Incognito mode; see chrome::GetBrowserContextRedirectedInIncognito() in the incognito_helpers.cc. 

I don´t think we have support for that now and not sure we are interested at this moment. Additionally, in latest versions of chrome the webDataServiceFactory doesn't override the GetBrowserContextToUse any more.

We could just remove this method, but I think it's clearer to have it for now, with that comment, just in case there is some Content API that uses it for some reason, so we do know we have this pending to be implemented.